font-patcher

```diff
@@ -633,20 +633,33 @@ class font_patcher:
         self.font_dim['height'] = abs(self.font_dim['ymin']) + self.font_dim['ymax']
 
 
-    def get_scale_factor(self, sym_dim):
-        scale_ratio = 1
+    def get_scale_factors(self, sym_dim, stretch):
+        """ Get scale in x and y as tuple """
+        # It is possible to have empty glyphs, so we need to skip those.
+        if not sym_dim['width'] or not sym_dim['height']:
+            return (1.0, 1.0)
 
-        # We want to preserve x/y aspect ratio, so find biggest scale factor that allows symbol to fit
-        scale_ratio_x = self.font_dim['width'] / sym_dim['width']
+        # For monospaced fonts all chars need to be maximum 'one' space wide
+        target_width = self.font_dim['width']
+        scale_ratio_x = target_width / sym_dim['width']
 
         # font_dim['height'] represents total line height, keep our symbols sized based upon font's em
-        # NOTE: is this comment correct? font_dim['height'] isn't used here
-        scale_ratio_y = self.sourceFont.em / sym_dim['height']
-        if scale_ratio_x > scale_ratio_y:
-            scale_ratio = scale_ratio_y
+        # Use the font_dim['height'] only for explicit 'y' scaling (not 'pa')
+        target_height = self.sourceFont.em if stretch == 'pa' else self.font_dim['height']
```

Comment:
If #749 is pulled, this would result in this change here:

```diff
-# Use the font_dim['height'] only for explicit 'y' scaling (not 'pa')
-target_height = self.sourceFont.em if stretch == 'pa' else self.font_dim['height']
+target_height = self.font_dim['height']
```

Which I think should be done.